wgpu-hal/src/dx12/adapter.rs

```diff
@@ -151,8 +151,20 @@ impl super::Adapter {
             hr == 0 && features2.DepthBoundsTestSupported != 0
         };
 
-        //Note: `D3D12_FEATURE_D3D12_OPTIONS3::CastingFullyTypedFormatSupported` can be checked
-        // to know if we can skip "typeless" formats entirely.
+        // winapi doesn't have options 3 :(
+
+        // let casting_fully_typed_format_supported = {
+        //     let mut features3: d3d12_ty::D3D12_FEATURE_DATA_D3D12_OPTIONS3 =
+        //         unsafe { mem::zeroed() };
+        //     let hr = unsafe {
+        //         device.CheckFeatureSupport(
+        //             d3d12_ty::D3D12_FEATURE_D3D12_OPTIONS3,
+        //             &mut features3 as *mut _ as *mut _,
+        //             mem::size_of::<d3d12_ty::D3D12_FEATURE_DATA_D3D12_OPTIONS3>() as _,
+        //         )
+        //     };
+        //     hr == 0 && features3.CastingFullyTypedFormatSupported != 0
+        // };
```

Comment:
We can define options3 ourselves - winapi exposes all of it's definition macros.